sci-libs/cantera/cantera-2.4.0.ebuild

```diff
@@ -0,0 +1,144 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5,6,7}} )
+
+inherit desktop python-r1 scons-utils toolchain-funcs
+
+DESCRIPTION="Object-oriented tool suite for chemical kinetics, thermodynamics, and transport"
+HOMEPAGE="http://www.cantera.org"
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+SRC_URI="https://github.com/Cantera/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+IUSE="+cpp +cti fortran pch python static-libs test"
+
+## Python2 is required by '<scons-3.0' to work.
+## Python3 automatic detection is used by cantera before compilling
+## if python3_package isn't set no "n".
+
+REQUIRED_USE="
+	|| ( cpp python )
+	cti? ( ${PYTHON_REQUIRED_USE} )
+	fortran? ( cpp )
+	python? ( cti )
+	static-libs? ( cpp )
+	?? ( python_targets_python3_4 python_targets_python3_5 python_targets_python3_6 python_targets_python3_7 )
+	"
+
+RDEPEND="
+	python? (
+		dev-python/numpy[${PYTHON_USEDEP}]
+	)
+"
+
+# SCons >= 3.0 is recomended, but builds with older versions
+DEPEND="
+	${RDEPEND}
+	dev-cpp/eigen
+	dev-libs/boost
+	dev-libs/libfmt:0=
+	dev-util/scons
+	sci-libs/sundials
+	fortran? (
+		sci-libs/sundials[fortran]
+		sys-devel/gcc[fortran]
+	)
+	python? (
+		dev-python/3to2
+		dev-python/cython[${PYTHON_USEDEP}]
+	)
+	test? (
+		>=dev-cpp/gtest-1.8.0
+	)
+"
+
+PATCHES=( "${FILESDIR}/${PN}_${PV}_disable_debug_and_optimization.patch" )
+
+## Full list of configuration options of Cantera is presented here:
+## http://cantera.org/docs/sphinx/html/compiling/config-options.html
+
+scons_vars=()
+set_scons_vars() {
+	scons_vars=(
+		CC="$(tc-getCC)"
+		CXX="$(tc-getCXX)"
+		cc_flags="${CXXFLAGS}"
+		cxx_flags="-std=c++11"
+		debug="no"
+		FORTRANFLAGS="${CXXFLAGS}"
+		use_pch=$(usex pch)
+## In some cases other order can break the detection of right location of Boost: ##
+		system_fmt="y"
+		system_sundials="y"
+		system_eigen="y"
+		extra_inc_dirs="/usr/include/eigen3"
+	)
+	use test || scons_vars+=( googletest="none" )
+}
+
+scons_targets=()
+set_scons_targets() {
+	scons_targets=(
+		prefix="/usr"
+		stage_dir="${D%/}"
```

Comment:
Using `${D}` is disallowed before `src_install`.

Reply:
Done.

I think as currently src_configure() doesn't change or create any files it could be moved to src_compile(). But currently it is still here because it's more clearly.

At the same time the the reconfiguring of scons and rebuild some objects could be occurs as mentioned in [gentoo wiki scons](https://wiki.gentoo.org/wiki/SCons) if it's options are changed as it takes place currently in src_install(). So I ussume to move `escons build...` with it's parameters to `src_install()` because mentioning of `${D}` variable isn't allowed out of `src_install()` section.